app/controllers/concerns/authentication_concern.rb

```diff
@@ -35,6 +35,11 @@ def selected_cis2_org_is_registered?
       )
     end
 
+    def selected_cis2_workgroup_is_valid?
+      selected_cis2_nrbac_role.key?("workgroups") &&
+        "schoolagedimmunisations".in?(selected_cis2_nrbac_role["workgroups"])
```

Comment:
Should the magic string be in a constant at the top of the file, for obviousness?